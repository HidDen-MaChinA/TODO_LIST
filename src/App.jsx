import React ,{ useState , useEffect , } from "react";
// import "./App.css";
function App (){
    const [done,setDone]=useState([])
    const newDone = (e,shouldBeDone) =>{
        setDone([...done,shouldBeDone]);
    }
    return (
        <div className="wrapper">
            <TODO func={newDone}/>
            <DONE done={done}/>
        </div>
    )
}
export function DONE({done}){
    return (
        <div className="todo">
            {done.map((information,id)=>(<div key={information+id}>{information}</div>))}
        </div>
    )
}
export function TODO({func}){
    const [state,setState]=useState([])
    const [innerText,setInnerText]=useState("")
    const enterNewValue = () => {
        if(innerText===""){
            alert("should not be empty")
        }else{
        setState([...state,innerText]);
        setInnerText("")
        }
    }
    const eraseOneLine = (e,id) =>{
        const test=state;
        func(e,test[id]);
        test.splice(id,1);
        setState([...test]);
    }
    return (
        <div className="DONE" >
            <input type='text' onInput={(e)=>{setInnerText(e.target.value)}} value={innerText} onKeyDown={(e)=>{e.key==="Enter" && enterNewValue() }} placeholder="todo Here"></input>
            <List informations={state} eraseOneLine={eraseOneLine}/>
        </div>
    )
}
export function List({ informations , eraseOneLine }){
    return(
        informations.map((information,id)=>(<p key={information+id}><input type="checkbox" onInput={(e)=>{eraseOneLine(e,id)}}></input>{information}</p>))
    )
}
export default App;
