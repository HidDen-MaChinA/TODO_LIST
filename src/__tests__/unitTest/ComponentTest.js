import React from 'react';
import renderer from 'react-test-renderer';
import {TODO , List , DONE } from '../../App.jsx';

it("expect to enter value when click enter",()=>{
    let MockState = "";
    const setMockState = (str) => {MockState = str}
    const component = renderer.create(
        <TODO func={setMockState} />,
    );
    let thisComponent = component.toJSON();
    expect(thisComponent).toMatchSnapshot();
})
it("expect to print information from props",()=>{
    const MockInformation = ["first task","second task"];
    const eraseOneLine = () => {}
    const component = renderer.create(
        <List informations={MockInformation} eraseOneLine={eraseOneLine} />
    );
    expect(component.toJSON()).toMatchSnapshot();
})
it("expect to print done task in Done component",()=>{
    const MockInformation = ["first task","second task"];
    const component = renderer.create(
        <DONE done={MockInformation} />
    );
    expect(component.toJSON()).toMatchSnapshot();
})