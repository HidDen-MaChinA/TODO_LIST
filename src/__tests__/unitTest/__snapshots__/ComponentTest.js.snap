// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to enter value when click enter 1`] = `
<div
  className="DONE"
>
  <input
    onInput={[Function]}
    onKeyDown={[Function]}
    placeholder="todo Here"
    type="text"
    value=""
  />
</div>
`;

exports[`expect to print done task in Done component 1`] = `
<div
  className="todo"
>
  <div>
    first task
  </div>
  <div>
    second task
  </div>
</div>
`;

exports[`expect to print information from props 1`] = `
[
  <p>
    <input
      onInput={[Function]}
      type="checkbox"
    />
    first task
  </p>,
  <p>
    <input
      onInput={[Function]}
      type="checkbox"
    />
    second task
  </p>,
]
`;
